### PR TITLE
Remove some mutexes from Windows implementation

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
After #1064 we don't need as many mutexes in the Windows implementation.

### CODE REVIEW

**Comment by Thomasdezeeuw** [comment]
@dtacalau I don't like doing this but I know the `registry_behind_arc` test fails. I think there is a problem with locking going on, but I don't really have the tools on Windows to properly debug it. Could you take a quick look? If you can't find it quickly let me know and I'll continue debugging.

**Comment by dtacalau** [comment]
@Thomasdezeeuw no problem, let me have a look.

**Comment by dtacalau** [comment]
@Thomasdezeeuw problem is that when the test ```registry_behind_arc``` reached this line ``` poll.poll(&mut events, Some(Duration::from_millis(1000)))``` the update queue is empty, so there's no sockstate to retrieve events for.  ```    unsafe fn update_sockets_events(&self) -> io::Result<()> {
        let mut update_queue = self.update_queue.lock().unwrap();```  . The update queue should have had two sock states, one for listener and another one for stream. That's all I could find for now.

**Comment by dtacalau** [comment]
@Thomasdezeeuw on another note I thought we agreed not to have large PRs with multiple changes at once.  Maybe I'm missing something but these commits look like major refactoring to me, not just ```remove some mutexes```. Is this the right thing to do before the release when we have just stabilized the tests? Sorry if I don't have the full picture with what we're trying to achieve here.

https://github.com/tokio-rs/mio/pull/1154#issuecomment-559043943
```
Going forward I would prefer to focus the prs on single issues, prs like these take way too long to get merged even without all the additional commits added on. In the future let's focus on a single (smaller) change in prs and try to get them merged faster
```

File: src/sys/windows/selector.rs [hunk 1]
```diff
@@ -357,6 +357,16 @@ impl Selector {
         self.inner.register(socket, token, interests)
     }
 
+    pub(super) fn register2(
```
Comment (dtacalau) [inline src/sys/windows/selector.rs]: what's the purpose of (re)register2?

File: src/sys/windows/selector.rs [hunk 2]
```diff
@@ -348,6 +347,7 @@ impl Selector {
         self.inner.select(events, timeout)
     }
 
+    // FIXME: replace with `register2`.
```
Comment (dtacalau) [inline src/sys/windows/selector.rs]: why replace it with register2?

**Comment by Thomasdezeeuw** [comment]
> @Thomasdezeeuw on another note I thought we agreed not to have large PRs with multiple changes at once. Maybe I'm missing something but these commits look like major refactoring to me, not just `remove some mutexes`.

You're right about the large pr, I've split the parts in three commits (one for each net type), followed by 3 clean-up commits. I could have send each commit as a separate pr.

>  Is this the right thing to do before the release when we have just stabilized the tests? Sorry if I don't have the full picture with what we're trying to achieve here.

Pr #1180 only recently got merged and this allows us to simplify these implementations. Personally I feel this makes the entire Windows implementation simpler. Especially if you look at the last four commits of #1183 (which builds on top of this pr).

**Comment by dtacalau** [comment]
@Thomasdezeeuw All I'm saying is this PR is hard to follow and review. I think the whole process would become far mor easier for everyone if you'd replace this large PR with smaller PRs, maybe the root cause of the ```registry_behind_arc ``` bug will also pop out and we won't lose time debugging it. 

What do you say? We got to start from somewhere with the small PRs agreement, right?

**Comment by Thomasdezeeuw** [comment]
@dtacalau can you review by commit, I always try to make nice clean commits. I don't have time to create a pr for each commit.

**Comment by dtacalau** [comment]
> @dtacalau can you review by commit, I always try to make nice clean commits. I don't have time to create a pr for each commit.

@Thomasdezeeuw I'd prefer separate PRs, why not letting these changes for later when you have time for multiple PRs?

Reviewing by commit doesn't seem feasible, in this case there are changes added in one commit and removed in next commits, for instance register2.

Just my 5 cents on the large vs small PRs issue: Reviewing a large PR would cost me more time than needed reviewing a bunch of smaller PRs. So someone will have to spend more time eventually.

**Comment by Thomasdezeeuw** [comment]
I'll try to redo this at some point when I have some time.
